pkg/store/prometheus.go

```diff
@@ -220,18 +230,66 @@ func (p *PrometheusStore) Series(r *storepb.SeriesRequest, s storepb.Store_Serie
 	return p.handleStreamedPrometheusResponse(s, httpResp, queryPrometheusSpan, extLset)
 }
 
-func (p *PrometheusStore) handleSampledPrometheusResponse(s storepb.Store_SeriesServer, httpResp *http.Response, querySpan tracing.Span, extLset labels.Labels) error {
-	ctx := s.Context()
+func (p *PrometheusStore) queryPrometheus(s storepb.Store_SeriesServer, r *storepb.SeriesRequest) error {
+	step := math.Max(float64(r.QueryHints.StepMillis/1000), 30)
+	minTime := math.Max(float64(r.MinTime), float64(r.MaxTime-5*time.Minute.Milliseconds()))
```

Comment:
I feel like this part needs an explanation in the comments, it's unclear from reading the code what is happening.

I assume 5 minutes come from the default PromQL lookback interval? Do we need to adjust this interval depending on the original PromQL engine's lookback interval? 

Also, shouldn't this be `math.Min` because this right now effectively limits each query to `[MaxTime-5m; MaxTime]` if `MaxTime-MinTime > 5m`? :thinking:

Reply:
It looks like we can always use a query range, we just have to make sure the step is set to a positive value when the query  coming from the Querier is an instant. This simplifies the logic a lot and we don't have to do obscure modifications to the query parameters.